src/librustc_typeck/astconv.rs

```diff
@@ -1446,7 +1447,15 @@ impl<'a, 'gcx, 'tcx> Bounds<'tcx> {
         }
 
         for bound_trait_ref in &self.trait_bounds {
-            vec.push(bound_trait_ref.to_predicate());
+            if let Some(node_id) = tcx.hir.as_local_node_id(bound_trait_ref.def_id()) {
+                if let hir::ItemTraitAlias(..) = tcx.hir.expect_item(node_id).node {
```

Comment:
Hmm. This doesn't seem like the right place to be doing these sorts of tests. By the time we get to `ty::TraitRef`, aliases should already be expanded, I think -- or, alternatively, we should do the expansion during trait resolution. The latter seems better and more like the strategy we want long term (not sure if that's the direction you've been going, still reading into the PR). Either way, this bit of code should't have to change.

Reply:
Agreed (see below).